src/randoop/util/SimpleList.java

```diff
@@ -40,11 +41,11 @@
  * creation time) proportional to N, not to the length of the new sequence.
  * <p>
  * When extending a Sequence with a new statement, we store the old sequence's
- * statements plus the new statement in a OneMoreElementList, which takes up
+ * statements plus the new statement in a {@code OneMoreElementList}, which takes up
  * only 2 references in memory (and constant creation time).
  * <p>
  */
-public abstract class SimpleList<T> {
+public abstract class SimpleList<T> implements Iterable<T> {
```

Comment:
SimpleList is not descriptive of the point of this data structure.  (Like some other comments, you could fix this one after pushing the changes if you want, but let's not forget about them.)
The point of the list is that it is immutable and supports very efficient append and concatenation; the documentation should say that and maybe the name should reflect it as well.